test/extended/openstack/loadbalancers.go

```diff
@@ -54,101 +56,103 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb] The Openstack
-		o.Expect(pool.LBMethod).Should(o.Equal(lbMethod), "Unexpected LBMethod on Openstack Pool: %q", pool.LBMethod)
-
-		g.By("accessing the service 100 times from outside and storing the name of the pods answering")
-		results := make(map[string]int)
-		for i := 0; i < 100; i++ {
-			// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
-			podName, err := getPodNameThroughUdpLb(svcIp, svcPort, "hostname")
-			if err != nil {
-				e2e.Logf("Error detected while accessing the LoadBalancer service on try %d: %q", i, err)
-			} else {
-				results[podName]++
+			g.By("Creating Openshift resources")
+			labels := map[string]string{"app": "udp-lb-default-dep"}
+			depName := "udp-lb-default-dep"
+			svcName := "udp-lb-default-svc"
+			testDeployment := e2edeployment.NewDeployment(depName, 2, labels, "udp-test",
+				imageutils.GetE2EImage(imageutils.Agnhost), appsv1.RollingUpdateDeploymentStrategyType)
+			testDeployment.Spec.Template.Spec.SecurityContext = e2epod.GetRestrictedPodSecurityContext()
+			testDeployment.Spec.Template.Spec.Containers[0].SecurityContext = e2epod.GetRestrictedContainerSecurityContext()
+			testDeployment.Spec.Template.Spec.Containers[0].Args = []string{"netexec", "--udp-port=8081"}
+			testDeployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{{
+				ContainerPort: 8081,
+				Protocol:      "UDP",
+			}}
+			deployment, err := clientSet.AppsV1().Deployments(oc.Namespace()).Create(context.TODO(),
+				testDeployment, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			jig := e2eservice.NewTestJig(clientSet, oc.Namespace(), svcName)
+			jig.Labels = labels
+			svc, err := jig.CreateLoadBalancerService(5*time.Minute, func(svc *v1.Service) {
+				svc.Spec.Ports = []v1.ServicePort{{Protocol: v1.ProtocolUDP, Port: 8082, TargetPort: intstr.FromInt(8081)}}
+				svc.Spec.Selector = labels
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("Checks from openshift perspective")
+			o.Expect(svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]).ShouldNot(o.BeEmpty(),
+				"load-balancer-id annotation missing")
+			loadBalancerId := svc.GetAnnotations()["loadbalancer.openstack.org/load-balancer-id"]
+			o.Expect(svc.Status.LoadBalancer.Ingress[0].IP).ShouldNot(o.BeEmpty(), "FIP missing on svc Status")
+			o.Expect(svc.Spec.ExternalTrafficPolicy).Should(o.Equal(v1.ServiceExternalTrafficPolicyTypeCluster),
+				"Unexpected ExternalTrafficPolicy on svc specs")
+
+			g.By("Checks from openstack perspective")
+			lb, err := octavialoadbalancers.Get(loadBalancerClient, loadBalancerId).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(lb.Provider).Should(o.Equal(strings.ToLower(lbProviderUnderTest)), "Unexpected provider in the Openstack LoadBalancer")
+
+			svcIp := svc.Status.LoadBalancer.Ingress[0].IP
```

Comment:
ditto before line 106

Reply:
Well, the string will be empty in worst case so no problem there, but you should check for length of the Ingress array, otherwise if it's empty, you'll end up with some golang panic here.